### DISCUSSION TITLE
The Doris 2.0.3 BE node keep crashing during loading data.

### DISCUSSION THREAD

**Discussion by Amanwan** [apache/doris#30496] [A - General / Q&A]
具体的be.out报错日志如下：
Java HotSpot(TM) 64-Bit Server VM warning: You have loaded library /data/doris/be/lib/hadoop_hdfs/native/libhadoop.so.1.0.0 which might have disabled stack guard. The VM will try to fix the stack guard now.
It's highly recommended that you fix the library with 'execstack -c <libfile>', or link it with '-z noexecstack'.
*** Query id: 0-0 ***
*** tablet id: 0 ***
*** Aborted at 1706500125 (unix time) try "date -d @1706500125" if you are using GNU date ***
*** Current BE git commitID: 37d31a5 ***
*** SIGSEGV address not mapped to object (@0x0) received by PID 203129 (TID 204438 OR 0x7ee46998e700) from PID 0; stack trace: ***
 0# doris::signal::(anonymous namespace)::FailureSignalHandler(int, siginfo_t*, void*) at /root/src/doris-2.0/be/src/common/signal_handler.h:417
 1# os::Linux::chained_handler(int, siginfo*, void*) in /data/doris/java8/jre/lib/amd64/server/libjvm.so
 2# JVM_handle_linux_signal in /data/doris/java8/jre/lib/amd64/server/libjvm.so
 3# signalHandler(int, siginfo*, void*) in /data/doris/java8/jre/lib/amd64/server/libjvm.so
 4# 0x00007F01E21ED270 in /lib64/libc.so.6
 5# doris::segment_v2::SegmentWriter::_encode_seq_column(doris::vectorized::IOlapColumnDataAccessor const*, unsigned long, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >*) at /root/src/doris-2.0/be/src/olap/rowset/segment_v2/segment_writer.cpp:822
 6# doris::segment_v2::SegmentWriter::append_block(doris::vectorized::Block const*, unsigned long, unsigned long) at /root/src/doris-2.0/be/src/olap/rowset/segment_v2/segment_writer.cpp:761
 7# doris::BetaRowsetWriter::_do_add_block(doris::vectorized::Block const*, std::unique_ptr<doris::segment_v2::SegmentWriter, std::default_delete<doris::segment_v2::SegmentWriter> >*, unsigned long, unsigned long) at /root/src/doris-2.0/be/src/olap/rowset/beta_rowset_writer.cpp:394
 8# doris::BetaRowsetWriter::_add_block(doris::vectorized::Block const*, std::unique_ptr<doris::segment_v2::SegmentWriter, std::default_delete<doris::segment_v2::SegmentWriter> >*, doris::FlushContext const*) at /root/src/doris-2.0/be/src/olap/rowset/beta_rowset_writer.cpp:425
 9# doris::BetaRowsetWriter::add_block(doris::vectorized::Block const*) in /data/doris/be/lib/doris_be
10# doris::PushHandler::_convert_v2(std::shared_ptr<doris::Tablet>, std::shared_ptr<doris::Rowset>*, std::shared_ptr<doris::TabletSchema>, doris::PushType) at /root/src/doris-2.0/be/src/olap/push_handler.cpp:279
11# doris::PushHandler::_do_streaming_ingestion(std::shared_ptr<doris::Tablet>, doris::TPushReq const&, doris::PushType, std::vector<doris::TTabletInfo, std::allocator<doris::TTabletInfo> >*) at /root/src/doris-2.0/be/src/olap/push_handler.cpp:173
12# doris::PushHandler::process_streaming_ingestion(std::shared_ptr<doris::Tablet>, doris::TPushReq const&, doris::PushType, std::vector<doris::TTabletInfo, std::allocator<doris::TTabletInfo> >*) at /root/src/doris-2.0/be/src/olap/push_handler.cpp:89
13# doris::EngineBatchLoadTask::_push(doris::TPushReq const&, std::vector<doris::TTabletInfo, std::allocator<doris::TTabletInfo> >*) at /root/src/doris-2.0/be/src/olap/task/engine_batch_load_task.cpp:293
14# doris::EngineBatchLoadTask::_process() at /root/src/doris-2.0/be/src/olap/task/engine_batch_load_task.cpp:248
15# doris::EngineBatchLoadTask::execute() in /data/doris/be/lib/doris_be
16# doris::StorageEngine::execute_task(doris::EngineTask*) at /root/src/doris-2.0/be/src/olap/storage_engine.cpp:1245
17# doris::PushTaskPool::_push_worker_thread_callback() in /data/doris/be/lib/doris_be
18# doris::ThreadPool::dispatch_thread() in /data/doris/be/lib/doris_be
19# doris::Thread::supervise_thread(void*) at /root/src/doris-2.0/be/src/util/thread.cpp:499
20# start_thread in /lib64/libpthread.so.0
21# __clone in /lib64/libc.so.6

**Comment by Amanwan** [comment]
具体的be.INFO 日志如下：

I0129 14:56:55.419160 237861 push_handler.cpp:241] tablet=327634.2112344526.9444edf9ed7e2ae4-32148e770f428084, file path=hdfs://ns1/doris_prd_data/spark_load/jobs/81124/dw_ods_product_t_scu_sale_base_ddx_dnhxwvc_1706510378649/327659/V1.dw_ods_product_t_scu_sale_base_ddx_dnhxwvc_1706510378649.327614.327617.327615.4.2112344526.parquet, file size=207212424
I0129 14:56:55.419257 266830 task_worker_pool.cpp:263] successfully submit task|type=REALTIME_PUSH|signature=53512|queue_size=1
W0129 14:56:55.419384 266830 task_worker_pool.cpp:268] failed to register task|type=REALTIME_PUSH|signature=53519
I0129 14:56:55.419615 237866 task_worker_pool.cpp:1474] get push task. signature=53512, priority=NORMAL push_type=LOAD_V2
I0129 14:56:55.419659 237866 engine_batch_load_task.cpp:269] begin to process push.  transaction_id=53499 tablet_id=327626, version=-1
I0129 14:56:55.419667 237866 push_handler.cpp:81] begin to realtime push. tablet=327626.2112344526.c24e14513b2fc214-cbc16e9cd574a5a2, transaction_id=53499
I0129 14:56:55.419879 237866 push_handler.cpp:241] tablet=327626.2112344526.c24e14513b2fc214-cbc16e9cd574a5a2, file path=hdfs://ns1/doris_prd_data/spark_load/jobs/81124/dw_ods_product_t_scu_sale_base_ddx_dnhxwvc_1706510378649/327659/V1.dw_ods_product_t_scu_sale_base_ddx_dnhxwvc_1706510378649.327614.327617.327615.2.2112344526.parquet, file size=207104884
I0129 14:56:55.834187 236613 daemon.cpp:218] os physical memory 125.42 GB. process memory used 2.38 GB, limit 100.33 GB, soft limit 90.30 GB. sys available memory 101.02 GB, low water mark 1.60 GB, warning water mark 3.20 GB. Refresh interval memory growth 0 B
I0129 14:56:56.138428 236613 daemon.cpp:218] os physical memory 125.42 GB. process memory used 2.66 GB, limit 100.33 GB, soft limit 90.30 GB. sys available memory 100.94 GB, low water mark 1.60 GB, warning water mark 3.20 GB. Refresh interval memory growth 0 B
W0129 14:56:56.383013 237866 status.h:393] meet error status: [INTERNAL_ERROR]Couldn't deserialize thrift msg:
No more data to read.

	0#  doris::Status doris::deserialize_thrift_msg<tparquet::PageHeader>(unsigned char const*, unsigned int*, bool, tparquet::PageHeader*) at /home/zcp/repo_center/doris_release/doris/be/src/util/thrift_util.h:153
	1#  doris::vectorized::PageReader::next_page_header() at /home/zcp/repo_center/doris_release/doris/be/src/common/status.h:442
	2#  doris::vectorized::ColumnChunkReader::next_page() at /home/zcp/repo_center/doris_release/doris/be/src/common/status.h:442
	3#  doris::vectorized::ScalarColumnReader::read_column_data(COW<doris::vectorized::IColumn>::immutable_ptr<doris::vectorized::IColumn>&, std::shared_ptr<doris::vectorized::IDataType const>&, doris::vectorized::ColumnSelectVector&, unsigned long, unsigned long*, bool*, bool) at /home/zcp/repo_center/doris_release/doris/be/src/common/status.h:442
	4#  doris::vectorized::RowGroupReader::_read_column_data(doris::vectorized::Block*, std::vector<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >, std::allocator<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > > > const&, unsigned long, unsigned long*, bool*, doris::vectorized::ColumnSelectVector&) at /home/zcp/repo_center/doris_release/doris/be/src/vec/exec/format/parquet/vparquet_group_reader.cpp:412
	5#  doris::vectorized::RowGroupReader::next_batch(doris::vectorized::Block*, unsigned long, unsigned long*, bool*) at /home/zcp/repo_center/doris_release/doris/be/src/vec/exec/format/parquet/vparquet_group_reader.cpp:312
	6#  doris::vectorized::ParquetReader::get_next_block(doris::vectorized::Block*, unsigned long*, bool*) at /home/zcp/repo_center/doris_release/doris/be/src/common/status.h:437
	7#  doris::PushBrokerReader::next(doris::vectorized::Block*) at /home/zcp/repo_center/doris_release/doris/be/src/common/status.h:442
	8#  doris::PushHandler::_convert_v2(std::shared_ptr<doris::Tablet>, std::shared_ptr<doris::Rowset>*, std::shared_ptr<doris::TabletSchema>, doris::PushType) at /home/zcp/repo_center/doris_release/doris/be/src/common/status.h:354
	9#  doris::PushHandler::_do_streaming_ingestion(std::shared_ptr<doris::Tablet>, doris::TPushReq const&, doris::PushType, std::vector<doris::TTabletInfo, std::allocator<doris::TTabletInfo> >*) at /home/zcp/repo_center/doris_release/doris/be/src/common/status.h:354
	10# doris::PushHandler::process_streaming_ingestion(std::shared_ptr<doris::Tablet>, doris::TPushReq const&, doris::PushType, std::vector<doris::TTabletInfo, std::allocator<doris::TTabletInfo> >*) at /home/zcp/repo_center/doris_release/doris/be/src/common/status.h:354
	11# doris::EngineBatchLoadTask::_push(doris::TPushReq const&, std::vector<doris::TTabletInfo, std::allocator<doris::TTabletInfo> >*) at /home/zcp/repo_center/doris_release/doris/be/src/common/status.h:354
	12# doris::EngineBatchLoadTask::_process() at /home/zcp/repo_center/doris_release/doris/be/src/common/status.h:354
	13# doris::EngineBatchLoadTask::execute() at /home/zcp/repo_center/doris_release/doris/be/src/olap/task/engine_batch_load_task.cpp:0
	14# doris::StorageEngine::execute_task(doris::EngineTask*) at /home/zcp/repo_center/doris_release/doris/be/src/common/status.h:442
	15# doris::PushTaskPool::_push_worker_thread_callback() at /home/zcp/repo_center/doris_release/doris/be/src/agent/task_worker_pool.cpp:0
	16# doris::ThreadPool::dispatch_thread() at /home/zcp/repo_center/doris_release/doris/be/src/util/threadpool.cpp:0
	17# doris::Thread::supervise_thread(void*) at /var/local/ldb_toolchain/bin/../usr/include/pthread.h:562
	18# start_thread
	19# clone
W0129 14:56:56.383028 237861 status.h:393] meet error status: [INTERNAL_ERROR]Couldn't deserialize thrift msg:
No more data to read.

	0#  doris::Status doris::deserialize_thrift_msg<tparquet::PageHeader>(unsigned char const*, unsigned int*, bool, tparquet::PageHeader*) at /home/zcp/repo_center/doris_release/doris/be/src/util/thrift_util.h:153
	1#  doris::vectorized::PageReader::next_page_header() at /home/zcp/repo_center/doris_release/doris/be/src/common/status.h:442
	2#  doris::vectorized::ColumnChunkReader::next_page() at /home/zcp/repo_center/doris_release/doris/be/src/common/status.h:442
	3#  doris::vectorized::ScalarColumnReader::read_column_data(COW<doris::vectorized::IColumn>::immutable_ptr<doris::vectorized::IColumn>&, std::shared_ptr<doris::vectorized::IDataType const>&, doris::vectorized::ColumnSelectVector&, unsigned long, unsigned long*, bool*, bool) at /home/zcp/repo_center/doris_release/doris/be/src/common/status.h:442
	4#  doris::vectorized::RowGroupReader::_read_column_data(doris::vectorized::Block*, std::vector<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >, std::allocator<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > > > const&, unsigned long, unsigned long*, bool*, doris::vectorized::ColumnSelectVector&) at /home/zcp/repo_center/doris_release/doris/be/src/vec/exec/format/parquet/vparquet_group_reader.cpp:412
	5#  doris::vectorized::RowGroupReader::next_batch(doris::vectorized::Block*, unsigned long, unsigned long*, bool*) at /home/zcp/repo_center/doris_release/doris/be/src/vec/exec/format/parquet/vparquet_group_reader.cpp:312
	6#  doris::vectorized::ParquetReader::get_next_block(doris::vectorized::Block*, unsigned long*, bool*) at /home/zcp/repo_center/doris_release/doris/be/src/common/status.h:437
	7#  doris::PushBrokerReader::next(doris::vectorized::Block*) at /home/zcp/repo_center/doris_release/doris/be/src/common/status.h:442
	8#  doris::PushHandler::_convert_v2(std::shared_ptr<doris::Tablet>, std::shared_ptr<doris::Rowset>*, std::shared_ptr<doris::TabletSchema>, doris::PushType) at /home/zcp/repo_center/doris_release/doris/be/src/common/status.h:354
	9#  doris::PushHandler::_do_streaming_ingestion(std::shared_ptr<doris::Tablet>, doris::TPushReq const&, doris::PushType, std::vector<doris::TTabletInfo, std::allocator<doris::TTabletInfo> >*) at /home/zcp/repo_center/doris_release/doris/be/src/common/status.h:354
	10# doris::PushHandler::process_streaming_ingestion(std::shared_ptr<doris::Tablet>, doris::TPushReq const&, doris::PushType, std::vector<doris::TTabletInfo, std::allocator<doris::TTabletInfo> >*) at /home/zcp/repo_center/doris_release/doris/be/src/common/status.h:354
	11# doris::EngineBatchLoadTask::_push(doris::TPushReq const&, std::vector<doris::TTabletInfo, std::allocator<doris::TTabletInfo> >*) at /home/zcp/repo_center/doris_release/doris/be/src/common/status.h:354
	12# doris::EngineBatchLoadTask::_process() at /home/zcp/repo_center/doris_release/doris/be/src/common/status.h:354
	13# doris::EngineBatchLoadTask::execute() at /home/zcp/repo_center/doris_release/doris/be/src/olap/task/engine_batch_load_task.cpp:0
	14# doris::StorageEngine::execute_task(doris::EngineTask*) at /home/zcp/repo_center/doris_release/doris/be/src/common/status.h:442
	15# doris::PushTaskPool::_push_worker_thread_callback() at /home/zcp/repo_center/doris_release/doris/be/src/agent/task_worker_pool.cpp:0
	16# doris::ThreadPool::dispatch_thread() at /home/zcp/repo_center/doris_release/doris/be/src/util/threadpool.cpp:0
	17# doris::Thread::supervise_thread(void*) at /var/local/ldb_toolchain/bin/../usr/include/pthread.h:562
	18# start_thread
	19# clone

**Comment by LemonLiTree** [comment]
具体导入方式是什么？能否提供一个复现该问题的case？

**Reply by Amanwan** [reply]
好的，可以复现的case如下，使用的导入方式是spark load 
doris表建表语句：
CREATE TABLE IF NOT EXISTS dw.doris_table_name (
  dt int comment 'dt',
  f_scu_id int comment '',
  city_id int comment '',
  area_id int comment '',
  channel int comment '',
  id int comment '',
  price bigint comment '',
  is_online int comment '',
  is_shelf int comment '',
  pic string comment '',
  `desc` string comment '',
  f_create_user_id int comment '',
  new_over_time int comment '',
  c_t int comment '',
  u_t int comment '',
  price_type int comment '',
  qa_status int comment '',
  is_deleted tinyint comment '',
  ts string comment '',
  time_limit_type tinyint comment '',
  st_str string comment '',
  et_str string comment ''
) Unique KEY(dt, f_scu_id, city_id, area_id, channel) comment '' PARTITION BY RANGE(dt) () DISTRIBUTED BY HASH(f_scu_id) BUCKETS auto PROPERTIES (
  "replication_allocation" = "tag.location.default: 3",
  "enable_unique_key_merge_on_write" = "true",
  "function_column.sequence_col" = "dt"
);



hive外部表建表语句：CREATE EXTERNAL TABLE IF NOT EXISTS dw.hive_ods_table_name (
  `id` int ,
  `f_scu_id` int ,
  `city_id` int ,
  `area_id` int ,
  `price` bigint ,
  `is_online` int ,
  `is_shelf` int ,
  `pic` string ,
  `desc` string ,
  `f_create_user_id` int ,
  `new_over_time` int ,
  `c_t` int ,
  `u_t` int ,
  `price_type` int ,
  `channel` int ,
  `qa_status` int ,
  `is_deleted` tinyint ,
  `ts` string ,
  `time_limit_type` tinyint ,
  `st_str` string ,
  `et_str` string ,
  `dt` int
)
ENGINE=HIVE
PROPERTIES (
"database" = "database",
"table" = "table_name",
"hadoop.security.authentication" = "simple",
"hive.version" = "1.1.0",
"hive.metastore.uris" = "xxx"
);

... 添加分区p20230912 

ALTER TABLE dw.doris_table_name ADD PARTITION p20230912 VALUES [("20230912"),("20230913"));

Load语句：LOAD LABEL load_label ( DATA FROM TABLE hive_ods_table_name INTO TABLE doris_table_name PARTITION (p20230912) where dt>=20230912 and dt<=20230912 )  WITH RESOURCE 'spark'(
"spark.yarn.queue"="queue",
"spark.executor.cores"="2",
"spark.driver.memory"="4g",
"spark.executor.memoryOverhead"="2g",
"spark.sql.hive.convertMetastoreOrc"="true",
"spark.executor.instances"="30",
"spark.executor.memory"="18g") PROPERTIES ( "timeout" = "36000" );

**Reply by LemonLiTree** [reply]
试试 用catalog导入
